src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/decode/Parser.scala

```diff
@@ -33,9 +33,16 @@ private[scalasdk] trait Parser[A] extends Serializable {
 
   def parse(row: String): DecodeResult[A] = {
     val values = row.split("\t", -1)
-    val zipped = knownKeys.zipAll(values, UnknownKeyPlaceholder, ValueIsMissingPlaceholder)
-    val decoded = decoder(zipped)
-    decoded.map { decodedValue => generic.from(decodedValue) }
+    if (values.length == 1) {
+      Validated.Invalid(NonEmptyList.of(NonTSVPayload))
```

Comment:
For me it looks that in fact or error type is:

```
NonTSVPayload | ColumnNumberMismatch | NonEmptyList[InvalidValue]
```

And not

```
NonEmptyList[NonTSVPayload] | NonEmptyList[ColumnNumberMismatch] | NonEmptyList[InvalidValue]
```

I.e. we should never check a case of `NonEmptyList.of(NonTSVPayload, ColumnNumberMismatch(_))`

Am I missing something?